src/lib/evil/evil_stat.c

```diff
@@ -0,0 +1,126 @@
+          r_fstatat = stat(pathname, statbuf);
+        else
+          r_fstatat = stat(pathname, statbuf);
+        return r_fstatat;
+     }
+   else
+     {
+        char *pathbuf = NULL;
+        size_t pathbuf_size = 0;
+        DWORD copied = 0;
+        do
+        {
+           pathbuf_size += MAX_PATH;
+           pathbuf = realloc(pathbuf, pathbuf_size * sizeof(char));
+           copied = GetModuleFileName(NULL, pathbuf, pathbuf_size);
+        } while (copied >= pathbuf_size);
+       
+        pathbuf_size = copied;
+        pathbuf = realloc(pathbuf, pathbuf_size * sizeof(char)); 
+        copied = GetModuleFileName(NULL, pathbuf, pathbuf_size);
+
+        int size_str;
+        for (size_str = strlen(pathbuf); size_str >= 0; size_str -- )
+         {
+            if (pathbuf[size_str] == '\\')
+              {
+                 pathbuf[size_str + 1] = 0;
+                 break;
+              }
+         }
+
+        size_str = strlen(pathbuf) + strlen(pathname);
+
+        char *path_complete = malloc(sizeof(char) * size_str);
+
+        path_complete =  strcat(pathbuf, pathname);
+      
+
+        if (flags == AT_SYMLINK_NOFOLLOW)
+          {
+             r_fstatat = stat(path_complete, statbuf);
+          }
+        else
+          { 
+             r_fstatat = stat(path_complete, statbuf);
+          }
+        return r_fstatat;
+     }
+}
+
+EAPI int 
+fstatat64(int dirfd, const char *pathname, struct stat *statbuf, int flags)
+{
+   int r_fstatat;
+
+   if (pathname[2] == '\\')
+     {  
+        if (flags == AT_SYMLINK_NOFOLLOW)
+          r_fstatat = stat64(pathname, statbuf);
+        else
+          r_fstatat = stat64(pathname, statbuf);
+        return r_fstatat;
+     }
+   else
+     {
+        char *pathbuf = NULL;
+        size_t pathbuf_size = 0;
+        DWORD copied = 0;
+        do
+        {
+           pathbuf_size += MAX_PATH;
+           pathbuf = realloc(pathbuf, pathbuf_size * sizeof(char));
+           copied = GetModuleFileName(NULL, pathbuf, pathbuf_size);
+        } while (copied >= pathbuf_size);
+       
+        pathbuf_size = copied;
+        pathbuf = realloc(pathbuf, pathbuf_size * sizeof(char)); 
+        copied = GetModuleFileName(NULL, pathbuf, pathbuf_size);
+
+        int size_str;
+        for (size_str = strlen(pathbuf); size_str >= 0; size_str -- )
+         {
+            if (pathbuf[size_str] == '\\')
+              {
+                 pathbuf[size_str + 1] = 0;
+                 break;
+              }
+         }
+
+        size_str = strlen(pathbuf) + strlen(pathname);
+
+        char *path_complete = malloc(sizeof(char) * size_str);
+
+        path_complete =  strcat(pathbuf, pathname);
+      
+
+        if (flags == AT_SYMLINK_NOFOLLOW)
+          {
+             r_fstatat = stat64(path_complete, statbuf);
+          }
+        else
+          { 
+             r_fstatat = stat64(path_complete, statbuf);
+          }
+        return r_fstatat;
+     }
+}
```

Comment:
What's the difference in the implementation of `fstatat` and `fstatat64`?

Reply:
according to documentation:[http://www.qnx.com/developers/docs/7.0.0/index.html#com.qnx.doc.neutrino.lib_ref/topic/f/fstatat.html](url)
> "The fstatat() and fstatat64() functions obtain information about the file or directory referenced in path. This information  is placed in the structure located at the address indicated by buf. The fstatat64() function is a large-file support version of fstatat()."_

Reply:
I see, but in the Windows implementation they are basically the function, aren't they? If so, why duplicate the code?